backbone/src/main/java/org/researchstack/backbone/task/Task.java

```diff
@@ -29,7 +29,18 @@
  */
 public abstract class Task implements Serializable {
     private String identifier;
-    public List<Step> steps;
+    /**
+     *
+     * @return the associated step for the task, in order
+     */
+    public abstract List<Step> getSteps();
+
+    /**
+     * Clear all completed steps after editing ends and load new one
+     *
+     * @param list
+     */
+    public abstract void resetCompletedTask(List<String> list);
```

Comment:
Can you annotate the parameter `list` with `@NonNull` or whatever is appropriate? Thanks.